src/common/NetworkPrivate.cpp

```diff
@@ -54,360 +149,311 @@ QString NetworkData::getHash()
-            data->timer_->setSingleShot(true);
-            data->timer_->start(data->timeoutMS_);
-        }
+    QObject::connect(&requester, &NetworkRequester::requestUrl, worker,
+                     &NetworkTask::run);
 
-        auto *reply = [&]() -> QNetworkReply * {
-            switch (data->requestType_)
-            {
-                case NetworkRequestType::Get:
-                    return NetworkManager::accessManager.get(data->request_);
+    emit requester.requestUrl();
+}
 
-                case NetworkRequestType::Put:
-                    return NetworkManager::accessManager.put(data->request_,
-                                                             data->payload_);
+// First tried to load cached, then uncached.
+void loadCached(std::shared_ptr<NetworkData> &&data)
+{
+    QFile cachedFile(getPaths()->cacheDirectory() + "/" + data->getHash());
 
-                case NetworkRequestType::Delete:
-                    return NetworkManager::accessManager.deleteResource(
-                        data->request_);
+    if (!cachedFile.exists() || !cachedFile.open(QIODevice::ReadOnly))
+    {
+        // File didn't exist OR File could not be opened
+        loadUncached(std::move(data));
+        return;
+    }
 
-                case NetworkRequestType::Post:
-                    if (data->multiPartPayload_)
-                    {
-                        assert(data->payload_.isNull());
+    // XXX: check if bytes is empty?
+    QByteArray bytes = cachedFile.readAll();
 
-                        return NetworkManager::accessManager.post(
-                            data->request_, data->multiPartPayload_);
-                    }
-                    else
-                    {
-                        return NetworkManager::accessManager.post(
-                            data->request_, data->payload_);
-                    }
-                case NetworkRequestType::Patch:
-                    if (data->multiPartPayload_)
-                    {
-                        assert(data->payload_.isNull());
+    qCDebug(chatterinoHTTP).noquote() << data->typeString() << "[CACHED] 200"
+                                      << data->request_.url().toString();
 
-                        return NetworkManager::accessManager.sendCustomRequest(
-                            data->request_, "PATCH", data->multiPartPayload_);
-                    }
-                    else
+    data->emitSuccess(
+        {NetworkResult::NetworkError::NoError, QVariant(200), bytes});
+    data->emitFinally();
+}
+
+void load(std::shared_ptr<NetworkData> &&data)
+{
+    if (data->cache_)
+    {
+        std::ignore = QtConcurrent::run([data = std::move(data)]() mutable {
+            loadCached(std::move(data));
+        });
+    }
+    else
+    {
+        loadUncached(std::move(data));
+    }
+}
+
+void NetworkData::emitSuccess(NetworkResult &&result)
+{
+    if (!this->onSuccess_)
+    {
+        return;
+    }
+
+#ifdef CHATTERINO_WARN_SLOW_HTTP
```

Comment:
The execution path here is needlessly complex. The response timer can be put into the same lambda as the non-timer code path. I prefer the non-timer code path to be fully eliminated. The timer impact shouldn't be a lot more than calling `time()` twice.